eth/beacon/state_machines/base.py

```diff
@@ -193,9 +222,562 @@ def active_state(self) -> ActiveState:
     def get_active_state_class(cls) -> Type[ActiveState]:
         """
         Return the :class:`~eth.beacon.types.active_states.ActiveState` class that this
-        StateMachine uses for crystallized_state.
+        StateMachine uses for active_state.
         """
         if cls.active_state_class is None:
             raise AttributeError("No `active_state_class` has been set for this StateMachine")
         else:
             return cls.active_state_class
+
+    #
+    # AttestationRecord
+    #
+    @classmethod
+    def get_attestation_record_class(cls) -> Type[AttestationRecord]:
+        """
+        Return the :class:`~eth.beacon.types.attestation_records.AttestationRecord` class that this
+        StateMachine uses for the current fork version.
+        """
+        if cls.attestation_record_class is None:
+            raise AttributeError("No `attestation_record_class` has been set for this StateMachine")
+        else:
+            return cls.attestation_record_class
+
+    @classmethod
+    def create_signing_message(cls,
```

Comment:
This looks like it could be extracted to a standalone function (which will simplify this class which is quite large).  Or, at minimum, it can be made into a `staticmethod`